atom/browser/api/atom_api_session.cc

```diff
@@ -214,54 +214,76 @@ std::map<uint32_t, v8::Global<v8::Object>> g_sessions;
 void RunCallbackInUI(const base::Callback<void()>& callback) {
   base::PostTaskWithTraits(FROM_HERE, {BrowserThread::UI}, callback);
 }
+
 template <typename... T>
 void RunCallbackInUI(const base::Callback<void(T...)>& callback, T... result) {
   base::PostTaskWithTraits(FROM_HERE, {BrowserThread::UI},
                            base::BindOnce(callback, result...));
 }
 
+void ResolveOrRejectPromiseInUI(atom::util::Promise promise, int net_error) {
+  if (net_error != net::OK) {
+    std::string err_msg =
+        "Failed to retrieve cache backend: " + net::ErrorToString(net_error);
```

Comment:
Having a specific error message inside a generic helper doesn't look good, either have the error message simply

```
std::string err_msg = net::ErrorToString(net_error);
```

or inline this helper as a lamda.